src/nni_manager/training_service/remote_machine/extends/linuxCommands.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import { OsCommands } from "../osCommands";
+import { RemoteCommandResult } from "../remoteMachineData";
+
+class LinuxCommands extends OsCommands {
+    public createFolder(folderName: string, sharedFolder: boolean = false): string {
+        let command;
+        if (sharedFolder) {
+            command = `umask 0; mkdir -p '${folderName}'`;
+        } else {
+            command = `mkdir -p '${folderName}'`;
+        }
+        return command;
+    }
+
+    public allowPermission(isRecursive: boolean = false, ...folders: string[]): string {
```

Comment:
Could you check where this command is used? I think we can always create folders with appropriate permission, instead of changing permission after creation.

Reply:
These folders are not created at same place, and it looks related to multi-user on same machine. To avoid regression, it won't be changed this time.